.github/workflows/ci.yml

```diff
@@ -11,22 +12,58 @@ env:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    name: "ci-${{ matrix.os }}"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
 
     steps:
+      - name: Configure Git
+        run: git config --global core.autocrlf false
+
       - uses: actions/checkout@v4
 
       - uses: Swatinem/rust-cache@v2
 
       - name: Check formatting
+        if: runner.os == 'Linux'
         run: cargo fmt --all -- --check
 
       - name: Build
-        run: cargo build --verbose --all-targets --tests --examples
+        run: cargo build --verbose --all-targets --tests --examples --release
 
       - name: Lint
-        run: cargo clippy --all-targets --all-features -- -D warnings
-
+        if: runner.os == 'Linux'
+        run: cargo clippy --all-targets --all-features --release -- -D warnings
       - name: Run tests
-        run: cargo test
+        run: cargo test --release
+
+      - name: Zip up the release build (unix)
+        if: runner.os != 'Windows'
+        run: |
+          export TARGET_TUPLE=$( rustc -vV | sed -n 's|host: ||p' )
+          echo "TARGET_TUPLE=$TARGET_TUPLE" >> $GITHUB_ENV
+          cd target/release
+          zip -r ../../mf2lsp-$TARGET_TUPLE.zip mf2lsp
+
+      - name: Zip up the release build (windows)
+        if: runner.os == 'Windows'
+        run: |
+          $env:TARGET_TUPLE = $( rustc -vV | Select-String -Pattern 'host: ' ).ToString().Replace('host: ', '')
+          echo "TARGET_TUPLE=$TARGET_TUPLE" >> $env:GITHUB_ENV
+          cd target\release
+          Compress-Archive -Path mf2lsp.exe -DestinationPath ..\..\mf2lsp-$env:TARGET_TUPLE.zip
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: mf2lsp-${{ env.TARGET_TUPLE }}
+          path: mf2lsp-*.zip
 
+      - name: GitHub release
```

Comment:
Does each time this runs for a given tag just upload the artifact to the existing release?

Reply:
Yes